plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/BlacklistScanner.java

```diff
@@ -43,12 +43,12 @@ public void setDocumentLocator(Locator locator) {
   }
   
   /**
-   * Ensures parser always starts with an empty stack.
+   * Ensures parser always starts with an empty queue.
    */
   @Override
   public void startDocument() throws SAXException {
     this.blacklist = new ArrayDeque<>();
-    this.characterEncoding = new Locator2Impl(locator).getEncoding();
+    setCharacterEncoding(new Locator2Impl(locator).getEncoding());
```

Comment:
I think instead of creating a new Locator2Impl here, you should simply cast the initial Locator to Locator2. If locator is not a Locator2, then you can't get the encoding from this anyway.